File: solution/1704. Determine if String Halves Are Alike/1704.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        const int n = s.size();
        unordered_set<char> vowels{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        
        int cnt = 0;
        for (int i = 0; i < n / 2; ++i) {
            if (vowels.count(s[i])) ++cnt;
            if (vowels.count(s[i + n/2])) --cnt;
        }
        return cnt == 0;
    }
};

File: solution/1704. Determine if String Halves Are Alike/1704.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        m = len(s) // 2
        vowels = "aeiouAEIOU"
        cnt = 0
        for i in range(m):
            cnt += s[i] in vowels
            cnt -= s[i + m] in vowels
        return cnt == 0
